src/main/java/com/esri/arcgisruntime/toolkit/skins/FloorFilterSkin.java

```diff
@@ -0,0 +1,712 @@
+    skinnable.selectedFacilityProperty().addListener((observable, oldValue, newValue) -> {
+      // if the facility has levels show the levels UI
+      if (newValue != null && !newValue.getLevels().isEmpty()) {
+        isShowLevelsProperty.set(true);
+      } else {
+        isShowLevelsProperty.set(false);
+      }
+      facilitiesListView.getSelectionModel().select(newValue);
+    });
+
+    // configure the text field that filters facilities by name
+    facilitiesFilterTextField.setPromptText("Filter facilities by name");
+    facilitiesFilterTextField.textProperty().addListener(obs -> {
+      var selectedSite = skinnable.getSelectedSite();
+      String filter = facilitiesFilterTextField.getText();
+      if (filter == null || filter.length() == 0) {
+        if (selectedSite != null) {
+          filteredFacilities.setPredicate(facility -> facility.getSite() == selectedSite);
+        } else {
+          filteredFacilities.setPredicate(facility -> true);
+        }
+      } else {
+        if (!isAllSitesProperty.get()) {
+          filteredFacilities.setPredicate(facility -> facility.getName().toLowerCase().contains(filter.toLowerCase()) &&
+            facility.getSite() == selectedSite);
+        } else {
+          filteredFacilities.setPredicate(facility -> facility.getName().toLowerCase().contains(filter.toLowerCase()));
+        }
+      }
+    });
+
+    // set the filtered list on the list view and configure a cell factory to display the facility names
+    facilitiesListView.setItems(filteredFacilities);
+    facilitiesListView.setPlaceholder(new Label("No facilities found"));
+    facilitiesListView.setCellFactory(v -> {
+      ListCell<FloorFacility> cell = new ListCell<>() {
+        @Override
+        protected void updateItem(FloorFacility facility, boolean empty) {
+          super.updateItem(facility, empty);
+          if (empty || facility == null) {
+            setText(null);
+          } else {
+            if (isAllSitesProperty.get()) {
+              setText(facility.getName() + " (" + facility.getSite().getName() + ")");
+            } else {
+              setText(facility.getName());
+            }
+          }
+        }
+      };
+      // if an already selected facility is clicked again, still update the viewpoint. Improves user experience.
+      cell.setOnMouseClicked(e -> {
+        if (skinnable.getSelectedFacility() != null) {
+          skinnable.getGeoView().setViewpoint(new Viewpoint(skinnable.getSelectedFacility().getGeometry().getExtent()));
+        }
+        invalidated();
+      });
+      return cell;
+    });
+
+    // if the properties are not bound and the facility is not already selected, set the facility to the control
+    facilitiesListView.getSelectionModel().selectedItemProperty().addListener((observable, oldValue, newValue) -> {
+      if (newValue != skinnable.getSelectedFacility() && newValue != null && !skinnable.selectedSiteProperty().isBound()
```

Comment:
I'd put the `newValue != null` first to avoid calling `skinnable.getSelectedFacility()` needlessly.